code/modules/projectiles/guns/ego_gun.dm

```diff
@@ -15,6 +15,11 @@
 	var/autofire	//In Rounds per decisecond
 	var/equip_bonus
 
+	//Reload mechanics
+	var/shotsleft
```

Comment:
it'd be nice if max shots was an editable variable to allow for things such as weapons with different modes or something